### ISSUE TITLE
Figure.subfigures dont show/update correctly

### ISSUE THREAD

**Issue by lwelzel** [matplotlib/matplotlib#19947]
### Bug report

**Bug summary**
When using Figure.subfigures(n, m) for any n or m larger than 2 the resulting figure does not draw properly when shown. 
Instead only the subfigures at [0:2, 0] and [0:2, -1] are drawn properly.

**Code for reproduction**
```python
import matplotlib.pyplot as plt

fig = plt.figure()
sub_figs = fig.subfigures(3, 3)

colors = ["red", "blue", "magenta", "gold", "green", "grey", "orange", "pink", "navy", "lightblue", "black"]

for i, (color, subfig) in enumerate(zip(colors,sub_figs.flatten())):
    subfig.set_facecolor(color)
    subfig.suptitle(f"subfig: {i}")

plt.show()

```

######### for (n, m) figures
```python 
import matplotlib.pyplot as plt

fig = plt.figure()
sub_figs = fig.subfigures(9, 10)

for i, subfig in enumerate(sub_figs.flatten()):
    subfig.set_facecolor("grey")
    subfig.suptitle(f"subfig: {i}")

plt.show()

```
**Actual outcome**

![image](https://user-images.githubusercontent.com/29613344/114410466-ae3edb80-9bab-11eb-92a5-aa07e68f4265.png)

Only the subfigures at [0:2, 0] and [0:2, -1] are drawn properly for any (n, m) figure.

**Expected outcome**
3x3 grid with the proper face colors. Similar results for 2x3 or 3x2 or larger grids. 

**Matplotlib version**
  * Operating system: Fedora (UNIX)
  * Matplotlib version (`import matplotlib; print(matplotlib.__version__)`): 3.4.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://backend_interagg
  * Python version: 3.7
  * Jupyter version (if applicable): NA
  * Other libraries: NA

Both python and Matplotlib from conda with manual update to Matplotlib version 3.4.1
conda channel: conda-forge

**Comment by jklymak** [comment]
It works fine if you actually put something in the figures and use constrained_layout.  I think what is happening is that because there is nothing in the subfigures, they are being shrunk to nothing.  

```python
import matplotlib.pyplot as plt

fig = plt.figure(constrained_layout=True)
sub_figs = fig.subfigures(3, 3)

colors = ["red", "blue", "magenta", "gold", "green", "grey", "orange", "pink", "navy", "lightblue", "black"]

for i, (color, subfig) in enumerate(zip(colors,sub_figs.flatten())):
    subfig.set_facecolor(color)
    ax = subfig.subplots()
    subfig.suptitle(f"subfig: {i}")

plt.show()
```

**Comment by lwelzel** [comment]
Interesting, thanks!

**Comment by jklymak** [comment]
There is definitely a big bug though.  Should be easy to fix, but not until 3.4.2 is released.  Thanks for finding this.

**Comment by lwelzel** [comment]
Happy to help and thanks for working on matplotlib.
